### DISCUSSION TITLE
Is scikit-survival RSF compatible with sklearn's Recursive Feature Elimination?

### DISCUSSION THREAD

**Discussion by phaniparsa** [sebp/scikit-survival#164] [Q&A]
Hello All,

I'm wondering if scikit-survival is compatible with sklearn library or not? I need to implement Recursive Feature Elimination with Random Survival Forests, as part of a project. Would anyone of you happen to know this?

Thank you!

**Comment by sebp** [comment]
RFE requires the estimator to have a feature_importances_ attribute, 
which currently is not implemented for RandomSurvivalForest.

**Reply by phaniparsa** [reply]
Are there any plans to implement this? If yes, I'd like to work on this. Will you please let me know how I can contribute towards this package?

Thank you!

**Reply by sebp** [reply]
I'm hesitant to support it, as mentioned in issue #140. Moreover, it seems that sklearn is going to deprecate split-based feature importance scores for random forest models. In 0.22 they introduced `sklearn.inspection.permutation_importance` as an alternative.

**Reply by phaniparsa** [reply]
gotcha, thanks!

**Comment by Benfeitas** [comment]
hi @sebp  and @phaniparsa , thanks for the discussion. I arrived to this same question and I see why you do not intend on implementing `feature_importances_`. I'm thinking to do the RFECV manually, but I'm wondering if you have any updates regarding this issue?

**Reply by phaniparsa** [reply]
Hello @Benfeitas : I personally explored Permutation Combination Feature Importance module in ELI5 and Scikit-Learn packages. Scikit-survival RSF model object is compatible with both of these. I went ahead and wrote a custom code snippet to do 10-fold model fitting and permutation combination approach on the test set with n_repeats as 10. Later I aggregate the importance scores across the 10-folds to identify the top features. Further, at work we also added in a random variable to the dataset(normal distribution or poisson distribution, depending on the nature of the variables) to see how many times a feature was ranked above the noise variable across 10-folds. For doing CV we used stratified group k-fold function to generate train test indices. Please let me know if you need further information on this. Happy to help.

**Reply by sebp** [reply]
I would recommend the approach outlined by @phaniparsa. Either you use sklearn's  [permutation_importance](https://scikit-learn.org/stable/modules/generated/sklearn.inspection.permutation_importance.html#sklearn.inspection.permutation_importance) or eli5's [PermutationImportance](https://eli5.readthedocs.io/en/latest/autodocs/sklearn.html#eli5.sklearn.permutation_importance.PermutationImportance).
